pkg/operator/ceph/csi/cluster_config.go

```diff
@@ -183,12 +186,19 @@ func updateCsiClusterConfig(curr, clusterKey string, newCsiClusterConfigEntry *C
 				break
 			}
 			centry.Monitors = newCsiClusterConfigEntry.Monitors
+			// update subvolumegroup and cephfs netNamespaceFilePath only when either is specified
+			// while always updating kernel and fuse mount options.
 			if newCsiClusterConfigEntry.CephFS.SubvolumeGroup != "" || newCsiClusterConfigEntry.CephFS.NetNamespaceFilePath != "" {
 				centry.CephFS = newCsiClusterConfigEntry.CephFS
+			} else {
+				centry.CephFS.KernelMountOptions = newCsiClusterConfigEntry.CephFS.KernelMountOptions
```

Comment:
Above on line 167 there is a check for `newCsiClusterConfigEntry != nil`, but this usage of the var is not inside that check. Don't we need to check again for nil here?

Reply:
> Above on line 167 there is a check for `newCsiClusterConfigEntry != nil`, but this usage of the var is not inside that check. Don't we need to check again for nil here?

The check at line 183 handles the nil case and breaks the for loop before it reaches this line.